### PR TITLE
Replace custom mock generator with mockery

### PR DESCRIPTION
**Pull request by donutloop**
The custom-made mock generator is not really satisfying the real requirements of the mock concept therefore I think it will be better to replace the custom mock generator with mockery. I began to deeply integrate the mockery cmd into the Apikit.

@seseloos let me know what do you think about this early draft. In the case you willing to accept this fundamental change then I will finalize the feature.

Mockery ref: https://github.com/vektra/mockery
autogenerated mock ref: https://github.com/ExperienceOne/apikit/blob/89938c251d6135449e625c5c09f50f05985f3aaf/tests/api/mock_VisAdminClient.go

### CODE REVIEW

**Comment by donutloop** [comment]
@seseloos Please review this draft :)

**Comment by donutloop** [comment]
```
{\__/}
(●_●)
( >🌮 Want a taco?

{\__/}
(●_●)
🌮  <\ noo!! it's mine

{\__/}
(●_●)
( >🌮 ok ok you can have it
```

@seseloos

**Comment by donutloop** [comment]
@seseloos

**Comment by seseloos** [comment]
@donutloop The draft looks good to me and I think it is a good idea to exchange the custom-made mocking functionality with something more sophisticated like _mockery_.

One thing that should be brought back into the tools functionality is the opportunity for the user to decide whether to create mocks along with the client and server code or not.

Can you please Tell me what the open steps to finalize the draft are.

**Comment by donutloop** [comment]
@seseloos I need to clean up the code and readding these flags to the code. Overall it shouldn't take much time to convert the draft into a stable version.

**Comment by donutloop** [comment]
i also need to add documentation

**Comment by donutloop** [comment]
@seseloos ready for review

File: README.md [hunk 1]
```diff
@@ -153,15 +153,51 @@ cd myproject
...
 ```
 
-### Generate API server and client
+### Generate API server, client and mock client
 
-The `apikit generate <api.yaml> <dest.dir> <package>` command generates the client and server API based on an OpenAPIv2 definition. Use `--only-client` if the server component and `--only-server` if the client component is not needed.
+The `apikit generate <api.yaml> <dest.dir> <package> <flags>` 
+command generates the client, mock client and server API based on an OpenAPIv2 definition.
+
+* (Optional) Use flag `--only-client` to only generate the server component should be generated.
```
Comment (seseloos) [inline README.md:161]: @donutloop Please correct the description of the `--only-client` flag to the following.

'(Optional) Use flag `--only-client` to only generate the client components'

File: README.md [hunk 2]
```diff
@@ -153,15 +153,51 @@ cd myproject
...
+* (Optional) Use flag `--only-server` to only generate the client component should be generated.
```
Comment (seseloos) [inline README.md:162]: @donutloop Please correct the description of the `--only-server` flag to the following.

'(Optional) Use flag `--only-server` to only generate the server components'

File: go.mod [hunk 3]
```diff
@@ -29,13 +29,14 @@ require (
 	github.com/prometheus/client_golang v1.4.1
 	github.com/prometheus/procfs v0.0.10 // indirect
 	github.com/sirupsen/logrus v1.4.2
+	github.com/spf13/viper v1.7.0 // indirect
```
Comment (seseloos) [inline go.mod:32]: @donutloop Please clean up the go modules with `go mod tidy`

**Review by seseloos — CHANGES_REQUESTED** [review]
@donutloop I have added some comments with changes for the README and the go.mod that will be needed before I will merge the pull request.

File: README.md (hunk 1 above)
Comment (MarcelEdmundFranke) [reply README.md:161]: Makes sense

File: README.md (hunk 2 above)
Comment (MarcelEdmundFranke) [reply README.md:162]: Mmh yeah

File: go.mod (hunk 3 above)
Comment (MarcelEdmundFranke) [reply go.mod:32]: Ok I gonna clean up the mod files

File: README.md (hunk 1 above)
Comment (donutloop) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (donutloop) [reply README.md]: fixed

File: go.mod (hunk 3 above)
Comment (donutloop) [reply go.mod]: fixed

**Comment by donutloop** [comment]
@seseloos please review again

**Comment by donutloop** [comment]
@seseloos

**Review by seseloos — APPROVED** [review]

**Pull request merged**